hubblestack/extmods/grains/cloud_details.py

```diff
@@ -138,20 +148,47 @@ def _get_gcp_details():
-                                                            headers=gcp_header, timeout=3).json()
+            gcp_extra['cloud_name'] = requests.get('http://metadata.google.internal/computeMetadata/v1/instance/name',
+                                                   headers=gcp_header, timeout=3).text
+            gcp_extra['cloud_hostname'] = requests.get('http://metadata.google.internal/computeMetadata/v1/instance/hostname',
+                                                       headers=gcp_header, timeout=3).text
+            gcp_extra['cloud_zone'] = requests.get('http://metadata.google.internal/computeMetadata/v1/instance/zone',
+                                                   headers=gcp_header, timeout=3).text
+            gcp_extra['cloud_image'] = requests.get('http://metadata.google.internal/computeMetadata/v1/instance/image',
+                                                    headers=gcp_header, timeout=3).text
+            gcp_extra['cloud_machine_type'] = requests.get('http://metadata.google.internal/computeMetadata/v1/instance/machine-type',
+                                                           headers=gcp_header, timeout=3).text
+            gcp_extra['cloud_tags'] = requests.get('http://metadata.google.internal/computeMetadata/v1/instance/tags?recursive=true',
+                                                   headers=gcp_header, timeout=3).json()
+            interface_list = requests.get('http://metadata.google.internal/computeMetadata/v1/instance/network-interfaces/?recursive=true',
+                                          headers=gcp_header, timeout=3).json()
+            for counter, value in enumerate(interface_list):
+                grain_name_network = "cloud_interface_{0}_network".format(counter)
+                gcp_extra[grain_name_network] = value['network']
+
+                grain_name_ip = "cloud_interface_{0}_ip".format(counter)
+                gcp_extra[grain_name_ip] = value['ip']
+
+                grain_name_subnetmask = "cloud_interface_{0}_subnetmask".format(counter)
+                gcp_extra[grain_name_subnetmask] = value['subnetmask']
+
+                grain_name_mac = "cloud_interface_{0}_mac_address".format(counter)
+                gcp_extra[grain_name_mac] = value['mac']
+
+                grain_name_forwardedips = "cloud_interface_{0}_forwarded_ips".format(counter)
+                gcp_extra[grain_name_forwardedips] = ','.join(value['forwardedIps'])
+
+                grain_name_targetips = "cloud_interface_{0}_target_ips".format(counter)
+                gcp_extra[grain_name_targetips] = ','.join(value['targetInstanceIps'])
+
+                grain_name_accessconfig_external_ips = "cloud_interface_{0}_accessconfigs_external_ips".format(counter)
+                external_ips_list = [ dict['externalIp'] for dict in value['accessConfigs'] if 'externalIp' in dict ]
```

Comment:
Can you please use a different variable name besides `dict`? I don't like shadowing built-ins.